hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -838,37 +864,65 @@ bool MDMParser::powerOff(void)
     LOCK();
     bool ok = false;
     bool continue_cancel = false;
+    bool check_ri = false;
     if (_init && _pwr) {
         MDM_INFO("\r\n[ Modem::powerOff ] = = = = = = = = = = = = = =");
         if (_cancel_all_operations) {
             continue_cancel = true;
             resume(); // make sure we can use the AT parser
         }
         for (int i=0; i<3; i++) { // try 3 times
-            sendFormated("AT+CPWROFF\r\n");
-            int ret = waitFinalResp(NULL,NULL,40*1000);
-            if (RESP_OK == ret) {
-                _pwr = false;
-                // todo - add if these are automatically done on power down
-                //_activated = false;
-                //_attached = false;
-                // Give the modem some time to switch off cleanly
-                HAL_Delay_Milliseconds(1000);
-                ok = true;
+            if(!_atOk()) {
```

Comment:
If we only have RI_UC connected to V_INT on the SARA R410 then no way after the PWR_UC pulse to validate modem powered off if AT interface is blocked. Suggest putting the entire section behind the DEV_SARA_410 check (i.e. `if(!_atOK() && _dev.dev == DEV_SARA_R410)`.

Reply:
It likely will power off though, and for now we could add a busy wait of 1s if we know it's not a R410 just to give it some time to do so (similar to the code we've had in there for CPWROFF).